src/main/java/com/google/firebase/database/util/EmulatorHelper.java

```diff
@@ -0,0 +1,71 @@
+package com.google.firebase.database.util;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Strings;
+import com.google.firebase.database.core.Path;
+import com.google.firebase.database.core.RepoInfo;
+import com.google.firebase.database.utilities.ParsedUrl;
+import com.google.firebase.database.utilities.Utilities;
+
+public final class EmulatorHelper {
+
+  private EmulatorHelper() {
+  }
+
+  @VisibleForTesting
+  public static final String FIREBASE_RTDB_EMULATOR_HOST_ENV_VAR = "FIREBASE_RTDB_EMULATOR_HOST";
+
+  private static String getEmulatorHostFromEnv() {
+    return System.getenv(FIREBASE_RTDB_EMULATOR_HOST_ENV_VAR);
+  }
+
+  public static ParsedUrl parsedUrlForEmulator(String dbName, String emulatorUrl) {
+    RepoInfo repoInfo = new RepoInfo();
+    repoInfo.host = emulatorUrl;
+    repoInfo.namespace = dbName;
+    repoInfo.secure = false;
+    ParsedUrl parsedUrl = new ParsedUrl();
+    parsedUrl.repoInfo = repoInfo;
+    parsedUrl.path = new Path("");
+    return parsedUrl;
+  }
+
+  @VisibleForTesting
+  static String extractEmulatorUrlFromDbUrl(String suppliedDatabaseUrl) {
+    if (Strings.isNullOrEmpty(suppliedDatabaseUrl)) {
+      return null;
+    }
+    ParsedUrl parsedUrl = Utilities.parseUrl(suppliedDatabaseUrl);
+    RepoInfo repoInfo = parsedUrl.repoInfo;
+    if (repoInfo.isSecure() || repoInfo.host.endsWith(".firebaseio.com") || !suppliedDatabaseUrl
```

Comment:
Is there a reason we disallow SSL? I would expect that a user could set it up to work properly.

Reply:
In theory a user can set up a reverse proxy that talks to the RTDB emulator. In practice, I cannot imagine many users doing this since we explicitly say that the emulator is not for production use, etc. I don't feel strongly about this as long as it is consistent either the Android SDK or the Python Admin SDK.

Reply:
Do we have more discussions here?

Reply:
The Android SDK supports talking to `https://localhost:8080`. I don't think we need to go out of our way to support SSL, but it seems non-ideal to write code that prevents SSL usage.

Reply:
Allowed this now.